### ISSUE TITLE
NoSuchElementException in RuntimeUpdatesProcessor if no tests are present in multi module in dev mode

### ISSUE THREAD

**Issue by vsevel** [quarkusio/quarkus#18941]
### Describe the bug

If a module contains no test, dev mode prints out the following exception while running the tests:

```
2021-07-22 18:15:21,978 ERROR [io.qua.test] (Test runner thread) Internal error running tests: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:148)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.lambda$setWatchedFilePaths$12(RuntimeUpdatesProcessor.java:948)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.setWatchedFilePathsInternal(RuntimeUpdatesProcessor.java:961)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.setWatchedFilePaths(RuntimeUpdatesProcessor.java:948)
        at io.quarkus.deployment.dev.testing.TestRunner.runInternal(TestRunner.java:252)
        at io.quarkus.deployment.dev.testing.TestRunner$1.run(TestRunner.java:126)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

The issue is with this code:
```
    public RuntimeUpdatesProcessor setWatchedFilePaths(Map<String, Boolean> watchedFilePaths, boolean isTest) {
        if (isTest) {
            setWatchedFilePathsInternal(watchedFilePaths, test, s -> Arrays.asList(s.getTest().get(), s.getMain()));
        }
```

In the described situation we have `s.getTest()` being _empty_. As such the `get()` will end up with the `NoSuchElementException`.

An easy workaround is to create at least a `src/main/java/.../EmptyTest.java`.

### Expected behavior

No exception

### Actual behavior

a `NoSuchElementException`. Note that this does not prevent the tests to run. It just adds noise.

### How to Reproduce?

- Create a multi-module application.
- Make sure you do not have any tests in one of the modules.
- run `mvn quarkus:dev -pl my-app`.
- Run the tests.

### Quarkus version or git rev

`2.0.1.Final`

**Comment by gsmet** [comment]
Can you test with 2.1.0.Final and see how it goes? (only the core artifacts have been released so use the quarkus-bom, 2.0.3.Final might also have a fix)

**Comment by stuartwdouglas** [comment]
Looks like a duplicate of https://github.com/quarkusio/quarkus/issues/18796

**Comment by vsevel** [comment]
> Looks like a duplicate of #18796, so should be fixed in 2.1.0.Final

indeed. thanks.
